### PR TITLE
refactor(scan): simplify write-in reason info type

### PR DESCRIPTION
**Pull request by eventualbuddha**
## Overview

Uses a named union type to reduce some of the duplication in this file.

## Demo Video or Screenshot
n/a

## Testing Plan 
n/a

## Checklist
- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.
- [x] I have added JSDoc comments to any newly introduced exports

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: frontends/bsd/src/screens/write_in_adjudication_screen.tsx [hunk 1]
```diff
@@ -45,6 +45,9 @@ import { Prose } from '../components/prose';
 import { Screen } from '../components/screen';
 import { AppContext } from '../contexts/app_context';
 
+type AnyWriteInAdjudicationReasonInfo =
+  | WriteInAdjudicationReasonInfo
+  | UnmarkedWriteInAdjudicationReasonInfo;
```
Comment (jonahkagan) [inline frontends/bsd/src/screens/write_in_adjudication_screen.tsx:50]: How about this? I think if I were using these types I might assume accidentally that `WriteInAdjudicationReasonInfo` is the main type

```suggestion
type WriteInAdjudicationReasonInfo =
  | MarkedWriteInAdjudicationReasonInfo
  | UnmarkedWriteInAdjudicationReasonInfo;
```

**Review by benadida — COMMENTED** [review]
lgtm, leaving this as comment so you get a chance to address @jonahkagan's suggestion without auto-merge :)

**Comment by eventualbuddha** [comment]
> lgtm, leaving this as comment so you get a chance to address @jonahkagan's suggestion without auto-merge :)

Disabled auto-merge. Feel free to approve if you think this would be good after doing the refactor @jonahkagan suggested.

**Review by benadida — APPROVED** [review]
cool, agree with @jonahkagan's comments, and otherwise lgtm

File: frontends/bsd/src/screens/write_in_adjudication_screen.tsx (hunk 1 above)
Comment (eventualbuddha) [reply frontends/bsd/src/screens/write_in_adjudication_screen.tsx]: I changed this to make a broader global change like you suggested.

**Pull request merged**